From 4101806bf73caf25c8ce4e455b154901da1fe788 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Wed, 3 May 2017 11:49:09 +0200
Subject: [PATCH] qt: Improve width of pinentryconfirm

* qt/pinentryconfirm.cpp (PinentryConfirm::showEvent): Add spacer
item for text width.

--
This fixes a pinentry-qt4 bug where part of the text might have
been hidden. And improves the layout for pinentry-qt5 where
the fingerprint will no longer be wordwrapped. Needs to be
done in the show event because only there we have the icon available.
---
 qt/pinentryconfirm.cpp | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/qt/pinentryconfirm.cpp b/qt/pinentryconfirm.cpp
index 8b59d9d..50c19a4 100644
--- a/qt/pinentryconfirm.cpp
+++ b/qt/pinentryconfirm.cpp
@@ -19,6 +19,9 @@
 #include "pinentryconfirm.h"
 #include "pinentrydialog.h"
 #include <QAbstractButton>
+#include <QGridLayout>
+#include <QSpacerItem>
+#include <QFontMetrics>
 
 PinentryConfirm::PinentryConfirm(Icon icon, int timeout, const QString &title,
                                  const QString &desc, StandardButtons buttons, QWidget *parent) :
@@ -44,6 +47,18 @@ bool PinentryConfirm::timedOut() const
 
 void PinentryConfirm::showEvent(QShowEvent *event)
 {
+    static bool resized;
+    if (!resized) {
+        QGridLayout* lay = dynamic_cast<QGridLayout*> (layout());
+        if (lay) {
+            QSize textSize = fontMetrics().size(Qt::TextExpandTabs, text(), fontMetrics().maxWidth());
+            QSpacerItem* horizontalSpacer = new QSpacerItem(textSize.width() + iconPixmap().width(),
+                                                            0, QSizePolicy::Minimum, QSizePolicy::Expanding);
+            lay->addItem(horizontalSpacer, lay->rowCount(), 1, 1, lay->columnCount() - 1);
+        }
+        resized = true;
+    }
+
     QDialog::showEvent(event);
     raiseWindow(this);
 }
-- 
2.8.0.rc3

